pkg/arvo/sys/vane/ford.hoon

```diff
@@ -4658,6 +4658,23 @@
       ::
       ?.  ?=([~ %success %scry *] zuse-scry-result)
         (wrap-error zuse-scry-result)
+      ::
+      ::  short-circuit to .pit during boot
+      ::
+      ::    This avoids needing to recompile the kernel if we're asked
+      ::    to build %hoon one the home desk, at revision 1 or 2.
```

Comment:
How does this deal with zuse changes where zuse.hoon differs in the arvo snapshot from the pill?

Reply:
You mean when `zuse.hoon` in the initial filesystem (`%into` event) in the pill is different from the `zuse.hoon` that was used to build %zuse in arvo (`%veer` event)? The answer to that is "not well". For now, I think  "don't make bad pills". Longer: it shouldn't be possible to get into this state, which is why arvo should keep a little mini filesystem in sync with the kernel. That's the approach from `neo/arvo`, and I've ported it and forward on a branch: https://github.com/urbit/urbit/blob/5d9c11e71243ebb9271b4ae9da4119a44b0762a9/pkg/arvo/gen/deep.hoon. I intend to implement that in arvo in my next round of changes.